src/Controls/tests/BindingSourceGen.UnitTests/BindingCodeWriterTests.cs

```diff
@@ -347,18 +347,46 @@ public void CorrectlyFormatsBindingWithIndexers()
     }
 
     [Fact]
-    public void CorrectlyFormatsBindingWithMemberAccessAndIndexAccess()
+    public void CorrectlyFormatsSimpleCast()
+    {
+        var generatedCode = BindingCodeWriter.BidningInterceptorCodeBuilder.GenerateConditionalPathAccess(
+            variableName: "source",
+            path: [
+                new Cast(new MemberAccess("A", IsNullable: true), TargetType: new TypeName("X", IsNullable: false, IsGenericParameter: false, IsValueType: false)),
+                new MemberAccess("B", IsNullable: false),
+            ],
+            depth: 2);
+
+        Assert.Equal("(source.A as X)?.B", generatedCode);
+    }
+
+    [Fact]
+    public void CorrectlyFormatsSimpleCastOfValueTypes()
+    {
+        var generatedCode = BindingCodeWriter.BidningInterceptorCodeBuilder.GenerateConditionalPathAccess(
+            variableName: "source",
+            path: [
+                new Cast(new MemberAccess("A", IsNullable: true), TargetType: new TypeName("X", IsNullable: false, IsGenericParameter: false, IsValueType: true)),
```

Comment:
Shouldn't TargetType have IsNullable: true as we are casting to nullable value type?

Reply:
I think you're right. If it wasn't an artificial scenario, those two properties won't end up in this configuration in a real-world case.

Reply:
What happens in the case we are casting sth nullable to reference type. The targetType should have isNullable: false?

Reply:
Do you have an C# example of what this would look like? It is possible that we will later drop this `isNullable` because we need just one to describe valid scenarios 🤷

Reply:
ValueType example:

```csharp
        label.SetBinding(Label.RotationProperty, static (Foo f) => (f.C as C?)?.X);

        public class Foo
        {
            public object? C { get; set; }
        }

        struct C
        {
            public int X { get; set; }
        }
```

ReferenceType example:

```csharp
        label.SetBinding(Label.RotationProperty, static (Foo f) => (f.C as C)?.X);

        public class Foo
        {
            public object? C { get; set; }
        }

        class C
        {
            public int X { get; set; }
        }
  ```

In ValueType example we cast to nullable C, in case of ReferenceType we cast to C

Reply:
Yes, in C# you can't do `as ValueType`, you always have to make it nullable `as ValueType?` so maybe it is unnecessary to even keep track of that flag?

I'll turn those examples into unit tests and I'll think about simplifying the record.

Reply:
I think we should rename `IsNullable` for the path access to `IsConditional` and keep `IsNullable` in `TypeName`. Maybe `TypeName` isn't a good name either, but both `Type` and `TypeInfo` are already defined in the BCL. `TypeDescription` maybe?

Reply:
I also don't think you can do `as ReferenceType?`. So isn't the information whether the underlying thing is a `ValueType` enough?

Reply:
I think that for casts we do not necessarily need the `isNullable` information. As we always cast to either nullable valueType or non-nullable reference type, right?

Reply:
I like `TypeDescriptor`. Im not sure about getting renaming of `isNullable` to `isConditional` in path access. I find `isNullable` quite clear, as in sth is nullable when it can be null. Also I found cases where we need to use the nullability of the last element in the path to determine the nullability of the `PropertyType`. So it makes sense that they are both the same. But it is not a strong opinion.